mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -645,6 +646,19 @@ def _full_path(self) -> str:
         assert path is not None
         return path
 
+    @noPosargs
+    @noKwargs
+    @FeatureNew('ExternalProgram.full_command', '1.6.1')
+    def full_command_method(self, args: T.List[TYPE_var], kwargs: TYPE_kwargs) -> str:
+        return self._full_command()
+
+    def _full_command(self) -> str:
+        if not self.found():
+            raise InterpreterException('Unable to get the path of a not-found external program')
+        cmd = self.held_object.get_command()
+        assert cmd is not None
+        return ' '.join(cmd)
```

Comment:
This fails badly if any command segment has spaces in it. Or any special character, really.

Reply:
I changed full_command to cmd_array